lightning/src/chain/keysinterface.rs

```diff
@@ -1130,12 +1167,103 @@ impl KeysInterface for KeysManager {
 	}
 
 	fn read_chan_signer(&self, reader: &[u8]) -> Result<Self::Signer, DecodeError> {
-		InMemorySigner::read(&mut io::Cursor::new(reader), self.get_node_secret())
+		InMemorySigner::read(&mut io::Cursor::new(reader), self.node_secret.clone())
 	}
 
-	fn sign_invoice(&self, hrp_bytes: &[u8], invoice_data: &[u5]) -> Result<RecoverableSignature, ()> {
+	fn sign_invoice(&self, hrp_bytes: &[u8], invoice_data: &[u5], invoice_type: Invoice) -> Result<RecoverableSignature, ()> {
 		let preimage = construct_invoice_preimage(&hrp_bytes, &invoice_data);
-		Ok(self.secp_ctx.sign_recoverable(&hash_to_message!(&Sha256::hash(&preimage)), &self.get_node_secret()))
+		let secret = match invoice_type {
+			Invoice::Bolt11 => self.get_node_secret(Invoice::Bolt11)?,
+			Invoice::Phantom => return Err(()),
+		};
+		Ok(self.secp_ctx.sign_recoverable(&hash_to_message!(&Sha256::hash(&preimage)), &secret))
+	}
+}
+
+/// Similar to [`KeysManager`], but allows the node using this struct to receive phantom node
+/// payments.
+///
+/// A phantom node payment is a payment made to a phantom invoice, which is an invoice that can be
+/// paid to one of multiple nodes. This works because the invoice officially pays to a fake node
+/// (the "phantom"), with route hints containing fake channel information that signals to LDK that a
+/// payment is officially destined for a phantom node.
```

Comment:
May be worth mentioning limitations and reasons here given this is what's linked to for "what are phantom payments" in a few places, eg how users will auto-retry on any offline nodes (and will route to the best node automagically), that you can't do MPP, etc